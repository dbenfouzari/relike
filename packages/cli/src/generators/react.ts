import { paramCase, snakeCase } from "change-case";
import fs from "fs";
import path from "path";

import { replace, replaceFileName } from "../utils/template";

/**
 * Generates component index file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateIndexFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateIndexFile = (key: string, componentName: string) => {
  const templateIndexPath = path.resolve(__dirname, "../templates/index/index.ts.txt");

  const templateIndex = fs.readFileSync(templateIndexPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateIndexPath, key, componentName)),
    content: replace(templateIndex, key, componentName, paramCase),
  };
};

/**
 * Generates component style file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateStyleFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateStyleFile = (key: string, componentName: string) => {
  const templateStylePath = path.resolve(__dirname, "../templates/react-style/__name__.module.scss.txt");

  const templateStyle = fs.readFileSync(templateStylePath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateStylePath, key, componentName)),
    content: replace(templateStyle, key, componentName, snakeCase),
  };
};

/**
 * Generates component story file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateStoryFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateStoryFile = (key: string, componentName: string) => {
  const templateStoryPath = path.resolve(__dirname, "../templates/react-story/__name__.stories.tsx.txt");

  const templateStory = fs.readFileSync(templateStoryPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateStoryPath, key, componentName)),
    content: replace(templateStory, key, componentName),
  };
};

/**
 * Generates component test file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateTestFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateTestFile = (key: string, componentName: string) => {
  const templateTestPath = path.resolve(__dirname, "../templates/react-test/__name__.test.tsx.txt");

  const templateTest = fs.readFileSync(templateTestPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateTestPath, key, componentName)),
    content: replace(templateTest, key, componentName),
  };
};

/**
 * Generates component file
 *
 * @param   {string}  key           The key to replace
 * @param   {string}  componentName The component name
 * @param   {boolean} hasStyle      Does the component have style
 * @example
 * generateComponentFile("name", name, flags.style);
 * @returns {any}                   File path and content.
 */
export const generateComponentFile = (key: string, componentName: string, hasStyle: boolean) => {
  const templateComponentPath = hasStyle
    ? path.resolve(__dirname, "../templates/react-component/__name__(if:style).tsx.txt")
    : path.resolve(__dirname, "../templates/react-component/__name__.tsx.txt");

  const templateComponent = fs.readFileSync(templateComponentPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateComponentPath, key, componentName)),
    content: replace(templateComponent, key, componentName),
  };
};
