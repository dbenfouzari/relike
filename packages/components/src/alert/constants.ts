import Color from "../color";
import Icons, { IconData } from "../icons";

export enum AlertSeverities {
  error = "error",
  warning = "warning",
  info = "info",
  success = "success",
}

export const DEFAULT_ICONS: Record<AlertSeverities, IconData> = {
  [AlertSeverities.error]: Icons.error_outline,
  [AlertSeverities.warning]: Icons.report_problem_outlined,
  [AlertSeverities.info]: Icons.info_outline,
  [AlertSeverities.success]: Icons.check_circle_outline,
};

export const DEFAULT_COLORS: Record<AlertSeverities, Record<"font" | "icon" | "background", Color>> = {
  [AlertSeverities.error]: {
    font: new Color(0xff5f2120),
    icon: new Color(0xffef5350),
    background: new Color(0xfffdeded),
  },
  [AlertSeverities.warning]: {
    font: new Color(0xff663c00),
    icon: new Color(0xffff9900),
    background: new Color(0xfffff4e5),
  },
  [AlertSeverities.info]: {
    font: new Color(0xff014361),
    icon: new Color(0xff03a9f4),
    background: new Color(0xffe5f6fd),
  },
  [AlertSeverities.success]: {
    font: new Color(0xff1e4620),
    icon: new Color(0xff4caf4f),
    background: new Color(0xffedf7ed),
  },
};

export type AvatarIcon = keyof typeof DEFAULT_ICONS;
export type AvatarColors = typeof DEFAULT_COLORS[keyof typeof DEFAULT_COLORS];
