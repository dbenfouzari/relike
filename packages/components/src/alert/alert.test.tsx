import { render, screen } from "@testing-library/react";

import Alert from "./alert";
import { AlertSeverities } from "./constants";

describe("Alert", () => {
  it("should render successfully", () => {
    render(<Alert />);

    expect(screen.getByRole("alert")).toMatchSnapshot();
  });

  it("should render with function children", () => {
    render(
      <Alert severity={AlertSeverities.error} title="Test">
        {({ color }) => (
          <div data-testid="content" style={{ backgroundColor: color.toRGB() }}>
            <h1>Hello World</h1>
          </div>
        )}
      </Alert>,
    );

    const style = window.getComputedStyle(screen.getByTestId("content"));

    expect(style.backgroundColor).toEqual("rgb(95, 33, 32)");
  });
});
