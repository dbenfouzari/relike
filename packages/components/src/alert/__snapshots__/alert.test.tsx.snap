// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render successfully 1`] = `
<div
  class="wrapper wrapper-0-2-1 wrapper-d0-0-2-2"
  role="alert"
>
  <div
    class="container-0-2-3 container-d0-0-2-4"
  >
    <div
      class="icon_wrapper"
    >
      <span
        class="jsx-3319743189 icon material-icons"
      >
        info_outline
      </span>
    </div>
    <div
      class="flex flex-0-2-5 flex-d0-0-2-6 message"
    >
      <div
        class="content"
      >
        <span
          class="elm-0-2-7 elm-d0-0-2-8"
        />
      </div>
    </div>
  </div>
</div>
`;
